### PR TITLE
PHP 8.2: Property in readonly class is readonly

### PR DESCRIPTION
**Pull request by kukulich**
Found by @ondrejmirtes

### CODE REVIEW

**Comment by kukulich** [comment]
I'm not sure if we should release 5.11.1.

**Comment by Ocramius** [comment]
If this is a bugfix, then yes, we should target `5.x`, then merge up

File: src/Reflection/ReflectionProperty.php [hunk 1]
```diff
@@ -207,7 +207,7 @@ public function isInitialized(object|null $object = null): bool
 
     public function isReadOnly(): bool
     {
-        return $this->node->isReadonly();
+        return $this->getDeclaringClass()->isReadOnly() || $this->node->isReadonly();
```
Comment (Ocramius) [inline src/Reflection/ReflectionProperty.php]: Minor optimization possible by reversing the operand order?

**Review by Ocramius — APPROVED** [review]

File: src/Reflection/ReflectionProperty.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/ReflectionProperty.php]: Done.

**Comment by Ocramius** [comment]
Github actions is having a hiccup here: unrelated to this patch, therefore :ship: 

Thanks @kukulich!

**Pull request merged**